db/migrate/20151014144004_host_collection_to_hosts.rb

```diff
@@ -0,0 +1,89 @@
+class HostCollectionToHosts < ActiveRecord::Migration
+  class Katello::System < ActiveRecord::Base
+    self.table_name = "katello_systems"
+  end
+
+  class Katello::HostCollection < ActiveRecord::Base
+    self.table_name = "katello_host_collections"
+  end
+
+  class Katello::SystemHostCollections < ActiveRecord::Base
+    self.table_name = "katello_system_host_collections"
+  end
+
+  class Katello::HostCollectionHosts < ActiveRecord::Base
+    self.table_name = "katello_host_collection_hosts"
+  end
+
+  def logger
+    Rails.logger
+  end
+
+  def create_host_collection_hosts_table
+    create_table :katello_host_collection_hosts, :force => true do |t|
+      t.references :host_collection
+      t.references :host
+      t.timestamps
+    end
+
+    add_foreign_key "katello_host_collection_hosts", "katello_host_collections",
+                    :name => "katello_host_collection_hosts_host_collection_id_fk", :column => "host_collection_id"
+
+    add_foreign_key "katello_host_collection_hosts", "hosts",
+                    :name => "katello_host_collection_hosts_host_id_fk", :column => "host_id"
+
+    add_index :katello_host_collection_hosts, [:host_collection_id]
+    add_index :katello_host_collection_hosts, [:host_id]
+  end
+
+  def create_system_host_collections
+    create_table :katello_system_host_collections, :force => true do |t|
+      t.integer  "system_id"
+      t.integer  "host_collection_id"
+      t.timestamps
+    end
+
+    add_foreign_key "katello_system_host_collections", "katello_host_collections",
+                    :name => "katello_system_host_collections_host_collection_id_fk", :column => "host_collection_id"
+
+    add_foreign_key "katello_system_host_collections", "katello_systems",
+                    :name => "katello_system_host_collections_host_id_fk", :column => "system_id"
+
+    add_index :katello_system_host_collections, [:host_collection_id]
+    add_index :katello_system_host_collections, [:system_id]
+  end
+
+  def up
+    create_host_collection_hosts_table
+
+    Katello::SystemHostCollections.all.each do |system_host_collection|
+      system = Katello::System.find(system_host_collection.system_id)
+      host_collection = Katello::HostCollection.find(system_host_collection.host_collection_id)
+
+      host_collection_host = Katello::HostCollectionHosts.new(:host_id => system.host_id, :host_collection_id => host_collection.id)
+      host_collection_host.save!
+    end
```

Comment:
You might could simplify this a bit by just doing something like:
```
Katello::HostCollection.each do |host_collection|
  host_collection.host_ids = host_collection.systems.pluck("host_id")
end
```

haven't tried it, but i think it would be faster